scoop/src/main/java/com/lyft/scoop/UiContainer.java

```diff
@@ -31,23 +34,36 @@ protected ViewControllerInflater getViewControllerInflater() {
         return new ViewControllerInflater();
     }
 
-    public void goTo(RouteChange screenChange) {
-        swap(screenChange);
+    public boolean onBack() {
+        return childCanGoBack();
     }
 
-    private void clean() {
-        active = null;
+    public void goTo(RouteChange routeChange) {
+        if (routeChangeQueue.isEmpty()) {
+            routeChangeQueue.add(routeChange);
 
-        this.removeAllViews();
+            swap(routeChange);
+        } else {
+            routeChangeQueue.add(routeChange);
+        }
     }
 
-    public boolean onBack() {
-        return childCanGoBack();
-    }
+    @Override
+    public void onTransitionCompleted() {
+        final TransitionListener transitionListener = getTransitionListener();
+        transitionListener.onTransitionCompleted();
+
+        if (!routeChangeQueue.isEmpty()) {
+            routeChangeQueue.pop();
```

Comment:
This could be executed, and if another thread enters the goTo method and pushes a routeChange onto the queue in the time it takes to get from line 57 to 58, two swaps would occur for the same route change.